### PR TITLE
Cypress automated test cases

### PR DESCRIPTION
**Pull request by deepankar-sajwan**
Test cases for login to portal based on user role

### CODE REVIEW

File: cypress/integration/login/login.spec.js [hunk 1]
```diff
@@ -1,7 +1,34 @@
 describe("Login", () => {
-  const username = Cypress.env("username");
-  const password = Cypress.env("password");
-  it("opens up the login page", () => {
-    cy.Login({ username, password })
+  it("logs in the default user", () => {
+    const user = Cypress.env("users").default;
+    cy.login(user);
+    cy.contains('Hello, there!');
+    cy.contains('Dashboard')
```
Comment (rathorevaibhav) [inline cypress/integration/login/login.spec.js:6]: let's remove this. We will eventually remove the dashboard text from the landing page for a person with no roles.

File: cypress/integration/login/login.spec.js [hunk 2]
```diff
@@ -1,7 +1,34 @@
...
+  it("logs in the employee user", () => {
+    const user = Cypress.env("users").employee;
+    cy.login(user);
+    cy.get("#wishlist").contains('Your wishlist');
+    cy.get('ul.navbar-nav').contains('CRM'); // employee should see CRM
+    cy.get('ul.navbar-nav').contains('KnowledgeCafe'); // employee should see KnowledgeCafe
+  });
+
+
+it("logs in the admin user", () => {
```
Comment (rathorevaibhav) [inline cypress/integration/login/login.spec.js:20]: please fix indentation

File: cypress/integration/login/login.spec.js [hunk 3]
```diff
@@ -1,7 +1,34 @@
...
+  const user = Cypress.env("users").admin;
+  cy.login(user);
+  cy.get('ul.navbar-nav').contains('CRM'); // employee should see CRM
```
Comment (rathorevaibhav) [inline cypress/integration/login/login.spec.js:23]: the comment requires an update here. please update at other places also
```suggestion
  cy.get('ul.navbar-nav').contains('CRM'); // admin should see CRM
```

File: cypress/integration/login/login.spec.js [hunk 4]
```diff
@@ -1,7 +1,34 @@
...
+  cy.get('ul.navbar-nav').contains('KnowledgeCafe'); // employee should see KnowledgeCafe
```
Comment (rathorevaibhav) [inline cypress/integration/login/login.spec.js:24]: please add check for other areas like Finance, HR for admins and super-admins

File: cypress/integration/login/login.spec.js [hunk 5]
```diff
@@ -1,7 +1,34 @@
...
+});
+
+it("logs in the superadmin user", () => {
+  const user = Cypress.env("users").superadmin;
+  cy.login(user);
+  cy.get('ul.navbar-nav').contains('CRM'); // employee should see CRM
+  cy.get('ul.navbar-nav').contains('KnowledgeCafe'); // employee should see KnowledgeCafe
```
Comment (rathorevaibhav) [inline cypress/integration/login/login.spec.js:31]: Superadmins should see the user management dropdown also. Please add that check.

File: cypress/integration/login/login.spec.js [hunk 6]
```diff
@@ -1,7 +1,34 @@
...
+
+
+it("logs in the admin user", () => {
+  const user = Cypress.env("users").admin;
+  cy.login(user);
+  cy.get('ul.navbar-nav').contains('CRM'); // employee should see CRM
+  cy.get('ul.navbar-nav').contains('KnowledgeCafe'); // employee should see KnowledgeCafe
+});
+
+it("logs in the superadmin user", () => {
```
Comment (rathorevaibhav) [inline cypress/integration/login/login.spec.js:27]: fix indentation

File: cypress.json.example [hunk 7]
```diff
@@ -1,9 +1,25 @@
 {
-  "baseUrl": "http://127.0.0.1:8000",
+  "baseUrl": "https://portal.test",
```
Comment (rathorevaibhav) [inline cypress.json.example:2]: I think changing this back to the original `baseUrl` will help the CI running
```suggestion
  "baseUrl": "http://127.0.0.1:8000",
```

**Review by rathorevaibhav — CHANGES_REQUESTED** [review]

File: cypress/integration/login/login.spec.js (hunk 1 above)
Comment (deepankar-sajwan) [reply cypress/integration/login/login.spec.js:6]: Done

File: cypress/integration/login/login.spec.js (hunk 2 above)
Comment (deepankar-sajwan) [reply cypress/integration/login/login.spec.js:20]: Done!

File: cypress/integration/login/login.spec.js (hunk 3 above)
Comment (deepankar-sajwan) [reply cypress/integration/login/login.spec.js:23]: done

File: cypress/integration/login/login.spec.js (hunk 6 above)
Comment (deepankar-sajwan) [reply cypress/integration/login/login.spec.js:27]: Done

File: cypress/integration/login/login.spec.js (hunk 4 above)
Comment (deepankar-sajwan) [reply cypress/integration/login/login.spec.js:24]: done

File: cypress/integration/login/login.spec.js (hunk 5 above)
Comment (deepankar-sajwan) [reply cypress/integration/login/login.spec.js:31]: done

**Review by rathorevaibhav — APPROVED** [review]

**Pull request merged**